integration-cli/docker_api_attach_test.go

```diff
@@ -99,128 +100,81 @@ func (s *DockerSuite) TestGetContainersWsAttachContainerNotFound(c *check.C) {
 
 func (s *DockerSuite) TestPostContainersAttach(c *check.C) {
 	testRequires(c, DaemonIsLinux)
-	out, _ := dockerCmd(c, "run", "-dit", "busybox", "cat")
-
-	r, w := io.Pipe()
-	defer r.Close()
-	defer w.Close()
-
-	conn, err := sockConn(time.Duration(10 * time.Second))
-	c.Assert(err, check.IsNil)
-
-	containerID := strings.TrimSpace(out)
 
-	req, err := http.NewRequest("POST", "/containers/"+containerID+"/attach?stream=1&stdin=1&stdout=1&stderr=1", bytes.NewReader([]byte{}))
-	c.Assert(err, check.IsNil)
-
-	client := httputil.NewClientConn(conn, nil)
-	defer client.Close()
-
-	// Do POST attach request
-	resp, err := client.Do(req)
-	c.Assert(resp.StatusCode, check.Equals, http.StatusOK)
-	// If we check the err, we get a ErrPersistEOF = &http.ProtocolError{ErrorString: "persistent connection closed"}
-	// This means that the remote requested this be the last request serviced, is this okay?
-
-	// Test read and write to the attached container
-	expected := []byte("hello")
-	actual := make([]byte, len(expected))
-
-	outChan := make(chan error)
-	go func() {
-		_, err := r.Read(actual)
-		outChan <- err
-		close(outChan)
-	}()
-
-	inChan := make(chan error)
-	go func() {
-		_, err := w.Write(expected)
```

Comment:
This is just testing the builtin `io.Pipe()` functions.
Write to one end and read from the other end, while it did nothing to do with the attach API.